import/bemobil_xdf2bids.m

```diff
@@ -417,124 +429,185 @@ function bemobil_xdf2bids(bemobil_config, numericalIDs, varargin)
-                eegcfg.elec = fullfile(participantDir, [bemobil_config.filename_prefix, num2str(participantNr) '_' bemobil_config.channel_locations_filename]); 
+            elseif isfield(bemobil_config, 'channel_locations_filename')
+                eegcfg.elec = fullfile(participantDir, [bemobil_config.filename_prefix, num2str(participantNr) '_' bemobil_config.channel_locations_filename]);
             end
-            
+
             % write eeg files in bids format
             data2bids(eegcfg, eeg);
-            
+
             for Ti = 1:numel(bemobil_config.other_data_types)
-                
+
                 switch bemobil_config.other_data_types{Ti}
-                    
+
                     case 'motion'
                         %--------------------------------------------------
                         %            Convert Motion Data to BIDS
                         %--------------------------------------------------
+                        
+                        % check for missing or wrong trackingsystem information
+                        if ~isfield(motionInfo.motion , 'tracksys')
+                            error ('Trackingsystems must be specified. Please create motionInfo.motion.tracksys containing names of tracking systems.') 
+                        elseif isempty (motionInfo.motion.tracksys)
+                            error ('Trackingsystems must be specified. Please enter name of tracking systems in motionInfo.motion.tracksys .') 
+                        else 
+                            if any(contains(motionInfo.motion.tracksys, '_'))
+                                error ('Name of trackingsystem is must not contain underscores. Please change name of trackingsystem.')
+                            end
+                        end
+                        
                         % check if any motion data was found at all
                         if isempty(xdfmotion)
-                            continue; 
+                            continue;
                         end
-                        
+
                         ftmotion = {};
-                        
+
                         % construct fieldtrip data
                         for iM = 1:numel(xdfmotion)
                             ftmotion{iM} = stream2ft(xdfmotion{iM});
                         end
-                        
+
                         % if needed, execute a custom function for any alteration to the data to address dataset specific issues
                         % (quat2eul conversion, unwrapping of angles, resampling, wrapping back to [pi, -pi], and concatenating for instance)
-                        motion = feval(motionCustom, ftmotion, motionStreamNames(bemobil_config.bids_rb_in_sessions(si,:)), participantNr, si, di);
-                        
+                        motion = feval(motionCustom, ftmotion{iM}, motionStreamNames(bemobil_config.bids_rb_in_sessions(si,:)), participantNr, si, di);
```

Comment:
This way only the final motion stream will be passed to the function. When there are multiple motion streams, this can be problematic.